Eplant/views/eFP/Viewer/index.tsx

```diff
@@ -467,4 +469,66 @@ export default class EFPViewer
       {props.geneticElement?.id}
     </Typography>
   )
+
+  citation = (props: { activeData?: EFPViewerData, state?: EFPViewerState, gene?: GeneticElement | null }) => {
+    const [xmlData, setXMLData] = useState<string[]>([])
+
+    const viewID = props.activeData?.views.find((v) => v.id == props.state?.activeView)?.name
+    const viewXML = props.activeData?.views.find((v) => v.id == props.state?.activeView)?.xmlURL
+    useEffect(() => {
+      const xmlLoad = async () => {
+        let xmlString = ""
+        if (viewXML) {
+          try {
+            const response = await fetch(viewXML)
+            const data = await response.text()
+            xmlString = data
+          } catch (error) {
+            console.error('Error fetching xmlData:', error)
+          }
+        }
+
+        // Extract <li> tags
+        if (xmlString !== "") {
+          const parser = new DOMParser();
+          const xmlDoc = parser.parseFromString(xmlString, 'text/xml');
+          const listItems = xmlDoc.querySelectorAll('info li');
+          const itemsArray = Array.from(listItems).map((liElement) => liElement.textContent ? liElement.textContent : "");
+          setXMLData(itemsArray)
+        } else {
+          setXMLData([])
+        }
+      }
+      xmlLoad()
+    }, [viewXML])
+
+    if (viewID) {
+      const citation = getCitation(viewID)
+      // Need to set inner HTML to capture nested HTML tags in some citations
+      return (
+        <div>
+          <div dangerouslySetInnerHTML={{ __html: citation.source }}></div>
+          <br></br><div dangerouslySetInnerHTML={{ __html: citation.notes }}></div>
+          <br></br>{citation.URL ? citation.URL : ""}
+          {
+            xmlData.length > 0 ?
+              (<ul>
+                {xmlData.map((item, index) => {
+                  if (item) {
+                    return (<li key={index}> {item}</li>)
+                  }
+                }
+                )}
+              </ul>
+              ) :
+              <></>
+          }
+          <br></br> This image was generated with the {viewID} at bar.utoronto.ca/eplant by Waese et al. 2017.
+          <Link href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" src="https://i.creativecommons.org/l/by/4.0/80x15.png" title="The ePlant output for your gene of interest is available under a Creative Commons Attribution 4.0 International License and may be freely used in publications etc." /></Link>
+        </div>
+      )
+    } else {
+      return <div>No Citation information provided.</div>
+    }
+  }
```

Comment:
It seems like every view will have mostly the same code here. Do you think it would be worth extracting this into its own component?

Reply:
Yeah good call